### PR TITLE
#450 Blog: Queue-Based Load Leveling Pattern 

### PR DESCRIPTION
**Pull request by Amarnath510**
I have added blog for Queue-Based Load Leveling Pattern. Please review the same.
There is one problem with images. I have added images to assets directory but still the images are not getting resolved. Please let me know how to resolve this issue.

### CODE REVIEW

**Comment by iluwatar** [comment]
The images show just fine on my local Jekyll. Did you check the instructions here https://github.com/iluwatar/java-design-patterns/wiki/07.-Working-with-the-web-site?

**Comment by iluwatar** [comment]
- In the first image Queue-load-level-pattern.png I would rename tasks -> requests and message queue -> task queue.
- In the second image Queue-load-level-scenario.png I'm not sure what is meant by "Web role instances". Could we talk about service requester role or something similar? Also as above, I would rename message queue -> task queue. Would it make sense to add multiple worker role instances in the picture to make it clear that there can be more than one?
- The implementation section is somewhat torso at the moment. I would add and explain some of the core classes of the example here.
- The author name is not displayed. Hopefully @markusmo3 can explain what is needed for the author name to show up correctly after the blog post. Do we have instruction for that?

**Comment by Amarnath510** [comment]
@iluwatar Thanks for the review.
I'm commenting on each of the points,
- Actually the images are picked up straight from the [post](https://msdn.microsoft.com/en-us/library/dn589783.aspx), so the names are the same. Do I need to create custom images?
- Same as above point.
- I will explain each class in detail.
- Not sure about this.

**Comment by iluwatar** [comment]
@Amarnath510 we cannot take images from another web site and use them on ours because of copyright issues. I recommend that you create the images for the blog yourself.

**Comment by Amarnath510** [comment]
@iluwatar I have updated the images and added description for the core classes.
Please review the same.

**Comment by markusmo3** [comment]
To add an author, first add your shortname (e.g. moe, ilu, ...) and your details [in the people.yml](https://github.com/iluwatar/java-design-patterns/blob/gh-pages/_data/people.yml)
then you can reference this shortname in your blogpost under the ['author' variable](https://github.com/iluwatar/java-design-patterns/pull/526/files#diff-9563b556dad7865eadd49c3843af6bf6R4)

**Comment by iluwatar** [comment]
@Amarnath510 looks good already. Great pictures! Here are some additional comments:
- The description of the classes is great, but can you also show the relevant code how it is implemented. For an example, see the how it is shown in the Hexagonal blog.
- As @markusmo3 instructed above please add the author information to the blog post

File: _config.yml [hunk 1]
```diff
@@ -8,8 +8,8 @@ description: Java Design Patterns
 
 # ! Attention: the lines starting with "baseurl" and "url" are edited on travis build with a search and replace regex
 # Any Change in those two lines should be checked with the "cibuild" script
-baseurl: "/java-design-patterns" # the subpath of your site, e.g. /blog/
-url: "https://iluwatar.github.io" # the base hostname & protocol for your site
+baseurl: "" # the subpath of your site, e.g. /blog/
+url: "http://localhost:4000" # the base hostname & protocol for your site
```
Comment (markusmo3) [inline _config.yml]: why this change? with `jekyll serve` in the java-design-patterns directory it should display the site correctly including all links, because they are relative. Jekyll serves the site under `http://127.0.0.1:4000/java-design-patterns/`

Comment (Amarnath510) [inline _config.yml]: @markusmo3 
My mistakes I have not reverted my local changes in the config file.
Reagarding the change, it was mentioned at [jekyll kick start](https://github.com/iluwatar/java-design-patterns/wiki/07.-Working-with-the-web-site) point 6. So I have changed it locally.

I will revert the changes and commit again.

**Comment by Amarnath510** [comment]
@iluwatar Okay I will check for typos once again.
But where to change the data? I don't see any blog post date in the blog? 
IMHO it should be the date we are publishing it right?

**Comment by iluwatar** [comment]
You can set the date by changing the filename of the post e.g. `_posts/queue-load-leveling/2016-12-9-queue-load-leveling.md`. We can use the current date or the date you want the post to go live. Let me know when you're ready with the final changes and I'll merge it.

**Comment by Amarnath510** [comment]
@iluwatar Thanks for the quick response. I have changed the blog post name and there are no typos.
Completed all the work .. I hope we can see the blog going live soon .. 👍 👍 👍

**Comment by iluwatar** [comment]
Well done @Amarnath510, it is now live at http://java-design-patterns.com/blog/. Looking forward to your further high quality contributions! :+1:

**Comment by Amarnath510** [comment]
👍 👍 👍 
@iluwatar Thank you so much.
